src/test/objectstore/store_test.cc

```diff
@@ -903,6 +903,205 @@ TEST_P(StoreTest, CompressionTest) {
   g_ceph_context->_conf->apply_changes(NULL);
 }
 
+TEST_P(StoreTest, garbageCollection) {
```

Comment:
You may use statistics returned by statfs method to measure amount of wasted data and verify it's decreasing (waste =  statfs.allocated - statfs.stored).